From c974bb0eaa204791c5abb7d051fe5b92ee018d80 Mon Sep 17 00:00:00 2001
From: Kobayashi <contact@zelinf.net>
Date: Tue, 4 Jan 2022 22:20:20 +0800
Subject: [PATCH] fix textDocument/selectionRange's method (#388)

* fix selection range

* remove Arbitrary instance for Value when aeson provides it
---
 lsp-types/src/Language/LSP/Types/Method.hs | 2 +-
 lsp/test/JsonSpec.hs                       | 3 +++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/lsp/test/JsonSpec.hs b/lsp/test/JsonSpec.hs
index 7d77e940..ab95dbf0 100644
--- a/lsp/test/JsonSpec.hs
+++ b/lsp/test/JsonSpec.hs
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP                  #-}
 {-# LANGUAGE UndecidableInstances #-}
 {-# LANGUAGE FlexibleContexts     #-}
 {-# LANGUAGE TypeInType           #-}
@@ -159,6 +160,7 @@ smallList = resize 3 . listOf
 instance (Arbitrary a) => Arbitrary (List a) where
   arbitrary = List <$> arbitrary
 
+#if !MIN_VERSION_aeson(2,0,3)
 instance Arbitrary J.Value where
   arbitrary = oneof
     [ J.String <$> arbitrary
@@ -166,6 +168,7 @@ instance Arbitrary J.Value where
     , J.Bool <$> arbitrary
     , pure J.Null
     ]
+#endif
 
 -- ---------------------------------------------------------------------
 
